umgc/spring2023
# Basic Tours Search

# Overview
Implements a basic search feature for tours in the back-end.  This search looks through tour names and descriptions.

# Testing
I recommend creating a few tours before testing so you can see how the filtering works.  I use [Postman ](https://www.postman.com/downloads/)to create tours currently.

This is my setup to create tours in Postman:
![image](https://user-images.githubusercontent.com/49327729/224559832-79040a47-770d-4ac2-985a-fac41d6c1c8e.png)
The POST URL will most likely be different on your device.  Check the URL that the Flask server gives you when it starts running and use that:
![image](https://user-images.githubusercontent.com/49327729/224559866-b28e56cd-2cfc-45ba-bda7-10b601d4ad3f.png)


1. Start up the flask server
2. Open the Flutter App
3. Choose the Search option from the Hamburger Menu
4. Try entering some text

![image](https://user-images.githubusercontent.com/49327729/224559708-ceac8818-ca12-442e-989b-8b19dd25878c.png)

Eventually, we will need to make clicking on a list item take the user to the correct tour.

# Comments
There has to be some data returned from the `/api/tours` endpoint for searching to be possible.  If there are no tours returns (and the status code is 200), there will be a popup saying "There are currently no tours to search for."  We can add another button in the future that lets the user go to the Create Tour page if we want.
